src/pages/home/sidebar/SidebarLinksData.tsx

```diff
@@ -1,149 +1,133 @@
-                message: PropTypes.string,
-
-                /** The transaction violation data */
-                data: PropTypes.shape({
-                    /** The transaction violation data field */
-                    field: PropTypes.string,
-
-                    /** The transaction violation data value */
-                    value: PropTypes.string,
-                }),
-            }),
-        ),
-    }),
+    transactionViolations: OnyxCollection<OnyxTypes.TransactionViolations>;
 
-    ...withCurrentUserPersonalDetailsPropTypes,
+    /** All policy members */
+    policyMembers: OnyxCollection<OnyxTypes.PolicyMembers>;
 };
 
-const defaultProps = {
-    chatReports: {},
-    isLoadingApp: true,
-    priorityMode: CONST.PRIORITY_MODE.DEFAULT,
-    betas: [],
-    policies: {},
-    policyMembers: {},
-    transactionViolations: {},
-    allReportActions: {},
-    ...withCurrentUserPersonalDetailsDefaultProps,
-};
+type SidebarLinksDataProps = CurrentReportIDContextValue &
+    SidebarLinksDataOnyxProps & {
+        /** Toggles the navigation menu open and closed */
+        onLinkClick: () => void;
+
+        /** Safe area insets required for mobile devices margins */
+        insets: EdgeInsets | undefined;
+    };
 
 function SidebarLinksData({
-    isFocused,
     allReportActions,
     betas,
     chatReports,
-    currentReportID,
     insets,
-    isLoadingApp,
+    isLoadingApp = true,
     onLinkClick,
     policies,
-    priorityMode,
-    network,
+    priorityMode = CONST.PRIORITY_MODE.DEFAULT,
     policyMembers,
     transactionViolations,
-    currentUserPersonalDetails,
-}) {
+    currentReportID,
+}: SidebarLinksDataProps) {
+    const {accountID} = useCurrentUserPersonalDetails();
+    const network = useNetwork();
+    const isFocused = useIsFocused();
     const styles = useThemeStyles();
     const {activeWorkspaceID} = useActiveWorkspace();
     const {translate} = useLocalize();
     const prevPriorityMode = usePrevious(priorityMode);
-
-    const policyMemberAccountIDs = getPolicyMembersByIdWithoutCurrentUser(policyMembers, activeWorkspaceID, currentUserPersonalDetails.accountID);
+    const policyMemberAccountIDs = getPolicyMembersByIdWithoutCurrentUser(policyMembers, activeWorkspaceID, accountID);
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    useEffect(() => Policy.openWorkspace(activeWorkspaceID, policyMemberAccountIDs), [activeWorkspaceID]);
+    useEffect(() => Policy.openWorkspace(activeWorkspaceID ?? '', policyMemberAccountIDs), [activeWorkspaceID]);
 
-    const reportIDsRef = useRef(null);
+    const reportIDsRef = useRef<string[] | null>(null);
     const isLoading = isLoadingApp;
-    const optionListItems = useMemo(() => {
+    const optionListItems: string[] | null = useMemo(() => {
         const reportIDs = SidebarUtils.getOrderedReportIDs(
             null,
-            chatReports,
+            chatReports as OnyxEntry<Record<string, OnyxTypes.Report>>,
```

Comment:
`SidebarUtils.getOrderedReportIDs` function is used in this page only, so I think it makes sense to update its typing instead of using assertion, wdyt?

Reply:
I am not sure if this is worth it because it will require a lot a changes in SidebarUtils also

Reply:
I agree, let's try at least 😅